packages/eslint-plugin-office-addins/src/rules/load-object-before-read.ts

```diff
@@ -0,0 +1,297 @@
+      return (node.parent?.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.parent.property.type === TSESTree.AST_NODE_TYPES.Identifier
+        && node.parent.property.name === "load");
+    }
+
+    function callsGetAPIFunction(node: TSESTree.Identifier): boolean {
+      return getFunctions.has(node.name);
+    }
+
+    function isVariableDeclaration(node: TSESTree.Node): boolean {
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.VariableDeclarator) {
+          return true;
+      }
+      return false;
+    }
+
+    function isGetVariableDeclaration(node: TSESTree.Node): boolean {
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.VariableDeclarator
+        && node.parent.init?.type === TSESTree.AST_NODE_TYPES.CallExpression
+        && node.parent.init.callee.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.parent.init.callee.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+          if(callsGetAPIFunction(node.parent.init.callee.property)) {
+            return true;
+          }
+      }
+      return false;
+    }
+
+    function isAssignmentExpression(node: TSESTree.Node): boolean {
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.AssignmentExpression) {
+          return true;
+      }
+      return false;
+    }
+
+    function isGetAssignmentExpression(node: TSESTree.Node): boolean {
+      if(node.parent?.type === TSESTree.AST_NODE_TYPES.AssignmentExpression
+        && node.parent.right.type === TSESTree.AST_NODE_TYPES.CallExpression
+        && node.parent.right.callee.type === TSESTree.AST_NODE_TYPES.MemberExpression
+        && node.parent.right.callee.property.type === TSESTree.AST_NODE_TYPES.Identifier) {
+          if(callsGetAPIFunction(node.parent.right.callee.property)) {
+            return true;
+          }
+      }
+      return false;
+    }
+
+    function getLoadedPropertyName(node: TSESTree.Node): string {
+      if(node.parent?.parent?.type === TSESTree.AST_NODE_TYPES.CallExpression
+        && node.parent.parent.arguments[0].type === TSESTree.AST_NODE_TYPES.Literal) {
+          return node.parent.parent.arguments[0].value as string;
+        }
+      return "error in getLoadedPropertyName";
+    }
+
+    function findLoadBeforeRead(scope: Scope) {
+      scope.variables.forEach((variable: Variable) => {
+        let loadLocation: Map <string, number> = new Map<string, number>();
+        let getFound: boolean = false;
+        variable.references.forEach((reference: Reference) => {
+          const node: TSESTree.Node = reference.identifier;
+          if (isVariableDeclaration(node)
+            || isAssignmentExpression(node)) {
+            if (isGetVariableDeclaration(node)
+              || isGetAssignmentExpression(node)) {
+                getFound = true;
+                return;
+            } else {
+              getFound = false;
+            }
+          }
+          
+          if(!getFound) {
+            return;
```

Comment:
If all the getFound is doing is determining return or not, you could just return from inside the if clause.